src/main/java/app/attestation/server/AttestationServer.java

```diff
@@ -1157,6 +1152,94 @@ public void handlePost(final HttpExchange exchange) throws IOException, SQLiteEx
         }
     }
 
+    private static class AttestationHistoryHandler extends PostHandler {
+        @Override
+        public void handlePost(final HttpExchange exchange) throws IOException, SQLiteException {
+            String fingerprint;
+            try (final JsonReader reader = Json.createReader(exchange.getRequestBody())) {
+                final JsonObject object = reader.readObject();
+                fingerprint = object.getString("fingerprint");
+                long offsetId = object.getJsonNumber("offsetId").longValue();
+                String requestTokenEncoded = object.getString("token");
+                final Account account = verifySession(exchange, false, requestTokenEncoded.getBytes());
+                if (account == null) {
+                    return;
+                }
+                if (!verifyDeviceFingerprint(account.userId, fingerprint)) {
+                    throw new NullPointerException("Invalid device fingerprint provided");
```

Comment:
This should report a proper error to the client like this:

```java
if (!verifyDeviceFingerprint(account.userId, fingerprint)) {
    exchange.sendResponseHeaders(403, -1);
    return;
}
```

Reply:
method has been optimized away.